packages/api/cms-api/src/dam/files/dam-media-alternatives/dam-media-alternative.resolver.ts

```diff
@@ -0,0 +1,159 @@
+import { FindOptions, Reference } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Type } from "@nestjs/common";
+import { Args, ID, Info, Mutation, Parent, Query, ResolveField, Resolver } from "@nestjs/graphql";
+import { GraphQLResolveInfo } from "graphql";
+
+import { CometValidationException } from "../../../common/errors/validation.exception";
+import { gqlArgsToMikroOrmQuery, searchToMikroOrmQuery } from "../../../common/filter/mikro-orm";
+import { extractGraphqlFields } from "../../../common/graphql/extract-graphql-fields";
+import { AffectedEntity } from "../../../user-permissions/decorators/affected-entity.decorator";
+import { RequiredPermission } from "../../../user-permissions/decorators/required-permission.decorator";
+import { FILE_ENTITY, FileInterface } from "../entities/file.entity";
+import { DamMediaAlternativeInput, DamMediaAlternativeUpdateInput } from "./dto/dam-media-alternative.input";
+import { DamMediaAlternativesArgs } from "./dto/dam-media-alternatives.args";
+import { PaginatedDamMediaAlternatives } from "./dto/paginated-dam-media-alternatives";
+import { DamMediaAlternative } from "./entities/dam-media-alternative.entity";
+
+export function createDamMediaAlternativeResolver({ File }: { File: Type<FileInterface> }): Type<unknown> {
+    @Resolver(() => DamMediaAlternative)
+    @RequiredPermission(["dam"], { skipScopeCheck: true })
```

Comment:
I'm not 100% sure if `skipScopeCheck` is okay here. The DamMediaAlternative doesn't have a scope but is attached to the scope of the linked files. Should I still check it here somehow?

Reply:
The problem is `damMediaAlternatives` and `createDamMediaAlternative` where I have to add a scope parameter that isn't used anywhere (because there is no AffectedEntity). Though I could check if the passed scope parameter is correct for `for` and `alternative`

Reply:
No wait, I missed something: I guess the scope check should be done via AffectedEntity for `for` and `alternative`. I'll do that